src/System.ServiceModel.Primitives/tests/ServiceModel/ChannelFactoryTest.cs

```diff
@@ -200,4 +201,31 @@ public static void CreateChannel_Of_Typed_Proxy_Using_BasicHttpBinding()
             }
         }
     }
+
+    [Fact]
+    public static void ChannelFactory_Async_Open_Close()
+    {
+        ChannelFactory<IRequestChannel> factory = null;
+        try
+        {
+            BasicHttpBinding binding = new BasicHttpBinding();
+
+            // Create the channel factory
+            factory = new ChannelFactory<IRequestChannel>(binding, new EndpointAddress(FakeAddress.HttpAddress));
+            Assert.True(CommunicationState.Created == factory.State, "The factory was not created.");
+
+            Task.Factory.FromAsync(factory.BeginOpen(null, null), factory.EndOpen).GetAwaiter().GetResult();
+            Assert.True(CommunicationState.Opened == factory.State, "The factory was not opened.");
+
```

Comment:
LGTM but I still have a nit observation.  The explanatory message should say what the CommunicationState value was if it was not the expected value. If this Assert fires in the lab, it will be critical to understand what the value was (Faulted?  Opening? Closing?).  As a rule of thumb, Assert messages should always say both what they expected to see and what they actually saw.